checker/src/main/java/org/checkerframework/checker/mustcall/MustCallTransfer.java

```diff
@@ -0,0 +1,375 @@
+import org.checkerframework.dataflow.analysis.TransferInput;
+import org.checkerframework.dataflow.analysis.TransferResult;
+import org.checkerframework.dataflow.cfg.node.LocalVariableNode;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.dataflow.cfg.node.ObjectCreationNode;
+import org.checkerframework.dataflow.cfg.node.StringConversionNode;
+import org.checkerframework.dataflow.cfg.node.TernaryExpressionNode;
+import org.checkerframework.dataflow.expression.JavaExpression;
+import org.checkerframework.dataflow.expression.Unknown;
+import org.checkerframework.framework.flow.CFAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFTransfer;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.GenericAnnotatedTypeFactory;
+import org.checkerframework.framework.util.JavaExpressionParseUtil.JavaExpressionParseException;
+import org.checkerframework.framework.util.StringToJavaExpression;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.TreePathUtil;
+import org.checkerframework.javacutil.TreeUtils;
+import org.checkerframework.javacutil.TypesUtils;
+import org.checkerframework.javacutil.trees.TreeBuilder;
+
+/**
+ * Transfer function for the must-call type system. Its primary purposes are (1) to create temporary
+ * variables for expressions (which allow those expressions to have refined information in the
+ * store, which the consistency checker can use), and (2) to reset refined information when a method
+ * annotated with @CreatesObligation is called.
+ */
+public class MustCallTransfer extends CFTransfer {
+
+  /** For building new AST nodes. */
+  private final TreeBuilder treeBuilder;
+
+  /** The type factory. */
+  private MustCallAnnotatedTypeFactory atypeFactory;
+
+  /**
+   * A cache for the default type for java.lang.String, to avoid needing to look it up for every
+   * implicit string conversion.
+   */
+  private @MonotonicNonNull AnnotationMirror defaultStringType;
+
+  /**
+   * Create a MustCallTransfer.
+   *
+   * @param analysis the analysis
+   */
+  public MustCallTransfer(CFAnalysis analysis) {
+    super(analysis);
+    atypeFactory = (MustCallAnnotatedTypeFactory) analysis.getTypeFactory();
+    ProcessingEnvironment env = atypeFactory.getChecker().getProcessingEnvironment();
+    treeBuilder = new TreeBuilder(env);
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitStringConversion(
+      StringConversionNode n, TransferInput<CFValue, CFStore> p) {
+    // Implicit String conversions should assume that the String's type is
+    // whatever the default for String is, not that the conversion is polymorphic.
+    TransferResult<CFValue, CFStore> result = super.visitStringConversion(n, p);
+    AnnotationMirror defaultStringType =
```

Comment:
`this.defaultStringType` is never set!  This sets a local variable that shadows the field.
Please abstract this logic into a method.
Or, how about setting the field unconditionally so that no check is needed at this use site?
Finally, if a method is retained, it can go inside the `if` statement below, so it will be executed less often.

Reply:
If the field were to be set unconditionally, we'd need to get a type mirror for `java.lang.String` from somewhere. It's more convenient to do it here, where there's already one definitely available in the form of the result of a string conversion (this avoids a reflective lookup).